packages/edit-site/src/components/site-hub/index.js

```diff
@@ -77,7 +77,8 @@ const SiteHub = memo(
 								variant="link"
 								href={ homeUrl }
 								target="_blank"
-								label={ __( 'View site (opens in a new tab)' ) }
+								icon={ external }
+								label={ decodeEntities( siteTitle ) }
```

Comment:
I'd suggest to entirely remove the label and use a `VisuallyHidden` text after the siteTitle to provide the info `(opens in a new tab)`. The visible text ant the actual accessible name willl be different but that's OK accorging to [WCAG Label in Name](https://www.w3.org/TR/WCAG22/#label-in-name).
See an example of this pattern at https://github.com/WordPress/gutenberg/blob/a1623c0216a58ce900cd0b56957789cec1606b68/packages/editor/src/components/preview-dropdown/index.js#L106-L111